### ISSUE TITLE
False Positives LC0029 - Use CompareDateTime method in Type Helper codeunit for DateTime variable comparisons.

### ISSUE THREAD

**Issue by tscottjendev** [StefanMaron/BusinessCentral.LinterCop#333]
This is a great rule but it only applies to Fields as the rounding is in SQL.  If I have DateTime variables I don't need dozens of these warnings when they will never be stored in SQL.  Is there a way to to restrict this to field symbols?

**Comment by Arthurvdv** [comment]
It would be great to have a way to only raise this rule when the DateTime variable has been read from SQL.

If the rule is restricted to only field symbols, the `MyDateTime` variable in the example below could still be effected by the rounding issue of SQL if I'm not mistaken?

```AL
procedure SomeDateTimeComparisons()
var
    Customer: Record Customer;
    MyDateTime: DateTime;
begin
    Customer.FindFirst();
    MyDateTime := Customer."Last Modified Date Time";
end;
```

If you have for example a codeunit where the DateTime variables will never be stored in SQL, you can suppress the rule for the whole object
```AL
#pragma warning disable LC0029
codeunit 50100 "MyCodeunit"
{
```

**Comment by tscottjendev** [comment]
I guess the assumption that all comparisons involving DateTime/Time variables have this issue seems a to be excessive.  Even in your example, the rounding has already occurred so there no need for any warning.  

I can use the `#pragma` to ignore them but it isn't my ideal solution in most cases.

**Comment by Arthurvdv** [comment]
I'm not sure if we're on the same page here. Just to clarify; My example will not raise the LC0029 rule, only when the `MyDateTime` variable would be compared to some other DateTime variable.

A possible solution would be to somehow figure out if the variable has been set from a field. This is very difficult to handle in a CodeCop and scenario's like variable by reference (VAR) and eventpublisher with a DateTime variable makes this limited to local variables only.

I'm following the conversation at [Add Method to Compare Time values #483](https://github.com/microsoft/BusinessCentralApps/issues/483), maybe this will bring new insights on how we can improve this rule in the future.

**Comment by tscottjendev** [comment]
I think we are I am just saying it badly.  I can use `#pragma` statements to resolve the places where the rule does not apply it.  I was just hoping for a more flexible rule interpretation.  If that is not possible/practical that is fine.
